### PR TITLE
Nullable expressions: undefined -> null

### PR DESCRIPTION
**Pull request by carterzenk**
- Changes expressions to use the `null` data type instead of `undefined`.
- This change caused some issues with the type snapshot checks.  Refactored the test snapshots to use `jest-runner-tsd`.  These tests now can check on the actual result set data type, rather than having to use the `GetDataType` class to extract nullability.
- Refactors the result set types to reduce duplication, and better explain that the `GetDataType` helper is now only used in `FromItemQuery`, rather than for testing only.
- Modifies the `toSql` helper to accept `Tokenable` types.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export type PickByValue<T, ValueType> = Pick<
   }[keyof T]
 >;
 
+export type Expand<T> = T extends infer O ? { [K in keyof O]: O[K] } : never;
```
Comment (cakoose) [inline src/types.ts:10]: Maybe a comment for what this does?

File: package.json [hunk 2]
```diff
@@ -33,23 +34,37 @@
     "prepublishOnly": "npm run build && npm test"
   },
   "jest": {
-    "moduleFileExtensions": [
-      "ts",
-      "js"
-    ],
-    "coveragePathIgnorePatterns": [
-      "__tests__",
```
Comment (cakoose) [inline package.json:41]: The old code used `__tests__` and `__checks__` folders. Can we keep that structure or is that hard?

(If we're starting out with the hope that this fork goes away eventually, it might be nice to avoid unnecessary changes.)

File: src/insert.ts [hunk 3]
```diff
@@ -327,7 +327,7 @@ export class InsertQuery<
               >
                 ? IsNotNull extends true
                   ? DataType | Expression<DataType, IsNotNull, any> | Query<any>
-                  : DataType | undefined | Expression<DataType, IsNotNull, any> | Query<any>
+                  : DataType | null | Expression<DataType, IsNotNull, any> | Query<any>
```
Comment (cakoose) [inline src/insert.ts:330]: Maybe worth defining a `type DbNull = null` somewhere and using that everywhere instead of the literal `null`?

(Right now, when I see `null`, I'm not sure if it represents a DB `null` value or if it's just one of the normal TS uses of `null`.)

File: package.json (hunk 2 above)
Comment (carterzenk) [reply package.json:41]: I moved this for less ignore configuration in these files, but if you'd prefer I can switch it back.

Comment (cakoose) [reply package.json:41]: Yeah, I think I'd prefer to leave it unchanged in this case, to stay close to the upstream code.

(I don't think that's _always_ the right answer, e.g. changing the type testing library like a change that's worth it.)

**Review by cakoose — COMMENTED** [review]
I commented on a thread.

**Review by cakoose — APPROVED** [review]

File: src/sql-functions.ts [hunk 4]
```diff
@@ -8,14 +8,21 @@ import {
...
-import type { Column, ColumnSet } from './column';
+import type { ColumnSet } from './column';
 import { DefaultExpression, Expression } from './expression';
 
 import { Query } from './query';
 import { Table } from './TableType';
 
+export interface Tokenable {
+    toTokens(): Token[];
+  }
```
Comment (cakoose) [inline src/sql-functions.ts:22]: Weird indentation...

I thought this project had a prettier?
